File: Foody/app/src/main/java/prateek_gupta/foody/data/database/entities/FoodJokeEntity.java
package prateek_gupta.foody.data.database.entities;

import androidx.room.Embedded;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

import prateek_gupta.foody.models.FoodJoke;
import prateek_gupta.foody.util.Constants;

@Entity(tableName = Constants.FOOD_JOKE_TABLE)
public class FoodJokeEntity {
    @PrimaryKey(autoGenerate = false)
    public Integer id = 0;

    @Embedded
    public FoodJoke foodJoke;

    public FoodJokeEntity(FoodJoke foodJoke) {
        this.foodJoke = foodJoke;
    }

    public FoodJoke getFoodJoke() {
        return foodJoke;
    }
}

File: Foody/app/src/main/java/prateek_gupta/foody/data/database/entities/FoodJokeEntity.kt
package prateek_gupta.foody.data.database.entities

import androidx.room.Embedded
import androidx.room.Entity
import androidx.room.PrimaryKey
import prateek_gupta.foody.models.FoodJoke
import prateek_gupta.foody.util.Constants.Companion.FOOD_JOKE_TABLE

@Entity(tableName = FOOD_JOKE_TABLE)
class FoodJokeEntity(
    @Embedded
    var foodJoke: FoodJoke
) {
    @PrimaryKey(autoGenerate = false)
    var id: Int = 0
}